crates/hathor-songs/src/database.rs

```diff
@@ -15,6 +23,38 @@ pub fn get_connection(db_path: &Path) -> Result<Connection, Box<dyn std::error::
     Ok(conn)
 }
 
+pub fn query_map_to_audiofile<ParamType>(
+    conn: &Connection,
+    sql: &str,
+    parameters: ParamType,
+) -> Result<Vec<AudioFile>, Box<dyn Error>>
+where
+    ParamType: Params,
+{
+    Ok(conn
+        .prepare(sql)?
+        .query_map(parameters, |row| {
+            // Literally impossible to return an error variant here.
+            // So instead we just unwrap and will panic and crash everything lol.
+            // TODO: Fix this dog water.
+            let song_path = PathBuf::from_str(&row.get::<_, String>(7)?).unwrap();
+            let img_path = PathBuf::from_str(&row.get::<_, String>(8)?).unwrap();
```

Comment:
@vablings Any idea how to address the comments here? I can't figure out how to return an error variant here if the path doesn't exist instead of unwrapping. Essentially at the moment, if a song exists in the DB and then the song gets moved/deleted, this query will panic rather than returning an error variant.

Reply:
Moved to #25